File: bitlap-core/src/main/kotlin/org/bitlap/core/sql/parser/BitlapSqlDdlNode.kt
/* Copyright (c) 2023 bitlap.org */
package org.bitlap.core.sql.parser

import org.apache.calcite.sql.SqlCall
import org.apache.calcite.sql.SqlNode
import org.apache.calcite.sql.SqlOperator
import org.apache.calcite.sql.parser.SqlParserPos
import org.bitlap.core.BitlapContext

/**
 * Mail: chk19940609@gmail.com
 * Created by IceMimosa
 * Date: 2021/7/28
 */
abstract class BitlapSqlDdlNode(
    open val pos: SqlParserPos,
    open val op: SqlOperator,
    open val operands: List<SqlNode>
) : SqlCall(pos), BitlapSqlDdlRel {

    protected val catalog = BitlapContext.catalog

    override fun getOperator(): SqlOperator = this.op
    override fun getOperandList(): List<SqlNode> = this.operands
}

File: bitlap-core/src/main/scala/org/bitlap/core/sql/parser/BitlapSqlDdlNode.scala
/**
 * Copyright (C) 2023 bitlap.org .
 */
package org.bitlap.core.sql.parser

import java.util.List as JList

import scala.jdk.CollectionConverters.*

import org.bitlap.core.BitlapContext
import org.bitlap.core.catalog.BitlapCatalog

import org.apache.calcite.sql.SqlCall
import org.apache.calcite.sql.SqlNode
import org.apache.calcite.sql.SqlOperator
import org.apache.calcite.sql.parser.SqlParserPos

abstract class BitlapSqlDdlNode(
  val _pos: SqlParserPos,
  override val op: SqlOperator,
  val operands: List[SqlNode])
    extends SqlCall(_pos)
    with BitlapSqlDdlRel {

  protected val catalog: BitlapCatalog = BitlapContext.catalog

  override def getOperator: SqlOperator       = this.op
  override def getOperandList: JList[SqlNode] = this.operands.asJava
}
